modules/deploy/pages/deployment-option/cloud/vpc-byo-gcp.adoc

```diff
@@ -0,0 +1,524 @@
+```
++
+.. Grant the host service agent the user role to the service account:
++
+```unset
+- gcloud projects add-iam-policy-binding <host-project-id> \
+    --member serviceAccount:service-<service-project-number>@container-engine-robot.iam.gserviceaccount.com \
+    --role roles/container.hostServiceAgentUser
+```
++
+.. Grant service project's GKE SA to manage firewall resources:
++
+```unset
+gcloud projects add-iam-policy-binding <host-project-id> \
+    --member=serviceAccount:service-<service-project-number>@container-engine-robot.iam.gserviceaccount.com \
+    --role=roles/compute.securityAdmin
+```
+For more information, see the https://cloud.google.com/kubernetes-engine/docs/how-to/cluster-shared-vpc#managing_firewall_resources[Google documentation^].
+
+. Create a cloud router and NAT at the host project. For example:
++
+```unset
+gcloud compute routers create <route-name> \
+   --project <host-project-id> \
+   --region <region> \
+   --network <shared-vpc-name>
+
+gcloud compute addresses create <address-name>
+
+gcloud compute routers nats create <nat-config-name> \
+   --project <host-project-id> \
+   --router <router-name> \
+   --region <region> \
+   --nat-all-subnet-ip-ranges \
+   --nat-external-ip-pool <address-name> \
+   --enable-endpoint-independent-mapping
+```
+
+. Create VPC firewall rules.
++
+** Redpanda ingress
+*** Target
+**** Tags redpanda-node
+*** Source
+**** IPv4 range (for example, 0.0.0.0/0 public access, otherwise RFC1918 and RFC6598: 10.0.0.0/8, 172.16.0.0/12, 192.168.0.0/16, 100.64.0.0/10)
+**** Tags redpanda
+*** Protocols and ports
+**** tcp:9092-9094
+**** tcp:30081
+**** tcp:30082
+**** tcp:30092
++
+** Master webhooks
+*** Target
+**** Source
+***** IPv4 range (for example, 10.0.7.240/28 or the GKE master CIDR range)
+**** Protocols and ports
+***** tcp:9443
+***** tcp:8443
+***** tcp:6443
+
+== Configure the service project
+
+. Enable GCP APIs in the service project. For example: 
++
+```unset
+gcloud services enable container.googleapis.com --project <service-project-id>
+```
++
+.Expand necessary APIs
+[%collapsible]
+====
+* cloudresourcemanager.googleapis.com
+* dns.googleapis.com
+* secretmanager.googleapis.com
+* compute.googleapis.com
+* iam.googleapis.com
+* storage-api.googleapis.com
+* container.googleapis.com
+* Serviceusage.googleapis.com
+====
+
+. Create storage buckets at the service project. Redpanda requires two storage buckets in the same region as the cluster:
++
+* Tiered Storage bucket: Redpanda uses Tiered Storage for writing log segments. The Tiered Storage bucket should not be versioned.
+* Management Storage bucket: Redpanda uses this bucket to store cluster metadata. The Management Storage bucket can have versioning enabled. 
+
+. Create service accounts at the service project.
+
+.. Redpanda agent SA
+
+... Redpanda agent custom role permissions
++
+.Expand necessary permissions
+[%collapsible]
+====
+* `compute.firewalls.get`
```

Comment:
Is there any way to programmatically inject this list? I'd love to be able to have the code and the documentation always be in sync.